swap/swap_test.go

```diff
@@ -301,6 +307,161 @@ func TestRepeatedBookings(t *testing.T) {
+					config.prvkey,
+					config.backendURL,
+					config.disconnectThreshold,
+					config.paymentThreshold,
+				)
+				if !strings.Contains(err.Error(), "no backend URL given") {
+					t.Fatal("no backendURL, but created SWAP")
+				}
+			},
+		},
+		{
+			name: "disconnect threshold lower than payment threshold",
+			configure: func(config *SWAPConfig) {
+				config.dbPath = dir
+				config.prvkey = prvKey
+				config.backendURL = ipcEndpoint
+				config.disconnectThreshold = DefaultDisconnectThreshold
+				config.paymentThreshold = DefaultDisconnectThreshold + 1
+			},
+			check: func(t *testing.T, config *SWAPConfig) {
+				_, err := NewSWAP(
+					config.dbPath,
+					config.prvkey,
+					config.backendURL,
+					config.disconnectThreshold,
+					config.paymentThreshold,
+				)
+				if !strings.Contains(err.Error(), "swap init error: disconnect threshold lower than payment threshold.") {
+					t.Fatal("disconnect threshold lower than payment threshold, but created SWAP")
+				}
+			},
+		},
+		{
+			name: "invalid backendURL",
+			configure: func(config *SWAPConfig) {
+				config.prvkey = prvKey
+				config.backendURL = "invalid backendURL"
+				config.disconnectThreshold = DefaultDisconnectThreshold
+				config.paymentThreshold = DefaultDisconnectThreshold
+			},
+			check: func(t *testing.T, config *SWAPConfig) {
+				defer os.RemoveAll(config.dbPath)
+				_, err := NewSWAP(
+					config.dbPath,
+					config.prvkey,
+					config.backendURL,
+					config.disconnectThreshold,
+					config.paymentThreshold,
+				)
+				if !strings.Contains(err.Error(), "swap init error: error connecting to Ethereum API") {
+					t.Fatal("invalid backendURL, but created SWAP", err)
+				}
+			},
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			dir, err := ioutil.TempDir("", "swarmSwap")
+			if err != nil {
+				t.Fatal(err)
+			}
+			defer os.RemoveAll(dir)
+			config.dbPath = dir
+
+			tc.configure(&config)
+			if tc.check != nil {
+				tc.check(t, &config)
+			}
+		})
+
+	}
+}
+
+//TestDisconnectThreshold tests that the disconnect threshold is reached when adding the DefaultDisconnectThreshold amount to the peers balance
+func TestDisconnectThreshold(t *testing.T) {
+	swap, clean := newTestSwap(t, ownerKey)
+	defer clean()
+	testPeer := newDummyPeer()
+	swap.addPeer(testPeer.Peer, swap.owner.address, swap.owner.Contract)
+	swap.Add(DefaultDisconnectThreshold, testPeer.Peer)
+	err := swap.Add(1, testPeer.Peer)
+	if !strings.Contains(err.Error(), "disconnect threshold") {
+		t.Fatal(err)
+	}
+}
+
+//TestPaymentThreshold tests that the payment threshold is reached when subtracting the DefaultPaymentThreshold amount from the peers balance
+func TestPaymentThreshold(t *testing.T) {
+	swap, clean := newTestSwap(t, ownerKey)
+	defer clean()
+	testPeer := newDummyPeer()
+	swap.addPeer(testPeer.Peer, swap.owner.address, swap.owner.Contract)
+	err := swap.Add(-DefaultPaymentThreshold, testPeer.Peer)
+	fmt.Println(err, 1)
+	// var cheque *Cheque
```

Comment:
ooops? Cleanup neeeded, also remove Println.

This test is not doing anything!

Reply:
Correct. Thanks for spotting!